packages/pyright-internal/src/analyzer/program.ts

```diff
@@ -1890,12 +1890,16 @@ export class Program {
                 return false;
             }
 
-            if (!this._disableChecker) {
+            if (!this._disableChecker && !fileToCheck.sourceFile.isFileDeleted()) {
                 // For ipython, make sure we check all its dependent files first since
                 // their results can affect this file's result.
                 const dependentFiles = this._checkDependentFiles(fileToCheck, chainedByList, token);
 
-                this._bindFile(fileToCheck);
+                this._bindFile(
+                    fileToCheck,
+                    undefined,
+                    /* skipFileNeededCheck */ fileToCheck.sourceFile.isBindingRequired()
```

Comment:
Why is this needed? I think I understand, but it's not obvious. Perhaps include a comment that explains why — to prevent it from being deleted in the future.

Reply:
Thanks. I'll add a comment.